graspy/align/seedless_procrustes.py

```diff
@@ -0,0 +1,359 @@
+
+        optimal_transport_num_reps : int (default=1000), optional
+            Number of repetitions in each iteration of the iterative optimal
+            transport problem. I.e. maximum number of repetitions in each
+            "E-step".
+
+        iterative_num_reps : int (default=100), optional
+            Number of reps in each iteration of the iterative optimal transport
+            problem. I.e. maxumum number of total iterations the whole "EM"
+            algorithm.
+
+        init : string, {'2d' (default), 'sign_flips', 'custom'}, optional
+
+            - '2d'
+                Uses :math:`2^d` different restarts, where :math:`d` is the
+                dimension of the datasets. In particular, tries all matrices
+                that are simultaneously diagonal and orthogonal. In other
+                words, these are diagonal matrices with all entries on the
+                diagonal being either +1 or -1. This is motivated by the fact
+                that spectral graph embeddings have two types of orthogonal
+                non-identifiability, one of which is captured by the orthogonal
+                diagonal matrices. The final result is picked based on the
+                final values of the objective function. For more on
+                this, see [2]_.
+            - 'sign_flips'
+                Initial alignment done by making the median value in each
+                dimension have the same sign. The motivation is similar to
+                that in '2d', except this is a heuristic that can save time,
+                but can sometimes yield suboptimal results.
+            - 'custom'
+                Expects either an initial guess for `self.Q_` or an initial
+                guess for `self.P_`, but not both. See `initial_Q` and
+                `initial_P`, respectively. If neither is provided,
+                initializes `initial_Q` to an identity with an appropriate
+                number of dimensions.
+
+        initial_Q : np.ndarray, shape (d, d) or None, optional (default=None)
+            An initial guess for the alignment matrix, `self.Q_`, if such
+            exists. Only one of `initial_Q`, `initial_P` can be provided at
+            the same time, and only if `init` argument is set to 'custom'. If
+            None, and `initial_P` is also None - initializes `initial_Q` to
+            identity matrix. Must be an orthogonal matrix, if provided.
+
+        initial_P : np.ndarray, shape (n, m) or None, optional (default=None)
+            Initial guess for the optimal transport matrix, `self.P`, if such
+            exists. Only one of `initial_Q`, `initial_P` can be provided at the
+            same time, and only if `init` argument is set to 'custom'. If None,
+            and `initial_Q` is also None - initializes `initial_Q` to identity
+            matrix. Must be a doubly stochastic matrix if provided (rows sum up
+            to 1/n, cols sum up to 1/m.)
```

Comment:
I thought defn was rows/cols sum to 1? https://en.wikipedia.org/wiki/Doubly_stochastic_matrix am i missing somethin

Reply:
i guess we don't mean doubly stochastic in the rigorous defn... so is there another name for this property?

Reply:
i've been confused as well, it feels like a stronger definition (because all "our" doubly stochastic matirces fit the wiki definition, but not the other way around). let's find out.

Reply:
i take that back, it's more like ours is kinda scaled.

Reply:
@bdpedigo change to "soft assignment matrix" everywhere?

Reply:
done